reference/configuration/twig.rst

```diff
@@ -105,6 +105,74 @@ Configuration
 
 .. _config-twig-exception-controller:
 
+auto_reload
+~~~~~~~~~~~
+
+**type**: ``boolean`` **default**: ``'%kernel.debug%'``
+
+If ``true``, whenever a template is rendered, Symfony checks first if its source
+code has changed since it was compiled. If it has changed, the template is
+compiled again automatically.
+
+autoescape_service
+~~~~~~~~~~~~~~~~~~
+
+**type**: ``string`` **default**: ``null``
+
+As of Twig 1.17, the escaping strategy applied by default to the template is
+determined during compilation time based on the filename of the template. This
+means for example that the contents of a ``*.html.twig`` template are escaped
+for HTML and the contents of ``*.js.twig`` are escaped for JavaScript.
+
+This option allows to define the Symfony service which will be used to determine
+the default escaping applied to the template.
+
+autoescape_service_method
+~~~~~~~~~~~~~~~~~~~~~~~~~
+
+**type**: ``string`` **default**: ``null``
+
+If ``autoescape_service`` option is defined, then this option defines the method
+called to determine the default escaping applied to the template.
+
+base_template_class
+~~~~~~~~~~~~~~~~~~~
+
+**type**: ``string`` **default**: ``'Twig_Template'``
+
+Twig templates are compiled into PHP classes before using them to render
+contents. This option defines the base class from which all the template classes
+extend. Using a custom base template is discouraged because it will make your
+application harder to maintain.
+
+cache
+~~~~~
+
+**type**: ``string`` **default**: ``'%kernel.cache_dir%/twig'``
+
+Before using the Twig templates to render some contents, they are compiled into
+regular PHP code. Compilation is a costly process, so the result is cached in
+the directory defined by this configuration option.
+
+Set this option to ``null`` to disable Twig template compilation. However, this
+is not recommended (even for the ``dev`` environment).
```

Comment:
We should maybe add a reference to the `auto_reload` option here so that one understand why you usually should not need to disable the cache.